wandb/data_types.py

```diff
@@ -1103,10 +1115,37 @@ class BoundingBoxes2D(JSONMetadata):
     Wandb class for 2D bounding Boxes
     """
 
+    def __init__(self, val, key, **kwargs):
+        super(BoundingBoxes2D, self).__init__(val)
+        self._val = val["box_data"]
+        self._key = key
+        # Add default class mapping
+        if not "class_labels" in val:
+            np = util.get_module("numpy", required="Semantic Segmentation mask support requires numpy")
+            classes = np.unique(list(map( lambda box: box["class_id"], val["box_data"]))).astype(np.int32).tolist()
+            class_labels = dict((c, "class_" + str(c)) for c in classes)
+            self._class_labels = class_labels
+        else:
+            self._class_labels = val["class_labels"]
+
+    def bind_to_run(self, run, key, step):
+        # bind_to_run key argument is the Image parent key
+        # the self._key value is the mask's sub key
+        super(BoundingBoxes2D, self).bind_to_run(run, key, step)
+        run._add_singleton("boundingBox/class_labels", key + "_wandb_delimeter_" + self._key , self._class_labels)
```

Comment:
mixed use of snake case and camel case here is a bit odd: "boundingBox/class_labels"
not sure if there is other examples where this is done or if this was intentional.

Reply:
I did it to match the python pattern of "class" camelcase. But, it's definitely weird here in the same instance. I changed it to all snake.